ai-ml/jupyterhub/addons.tf

```diff
@@ -0,0 +1,235 @@
+    })]
+  }
+}
+
+#---------------------------------------------------------------
+# EFS Filesystem for private volumes per user
+#---------------------------------------------------------------
+module "efs" {
+  source  = "terraform-aws-modules/efs/aws"
+  version = "~> 1.1"
+
+  creation_token = local.name
+  name           = local.name
+
+  # Mount targets / security group
+  mount_targets = {
+    for k, v in zipmap(local.azs, module.vpc.private_subnets) : k => { subnet_id = v }
+  }
+  security_group_description = "${local.name} EFS security group"
+  security_group_vpc_id      = module.vpc.vpc_id
+  security_group_rules = {
+    vpc = {
+      # relying on the defaults provdied for EFS/NFS (2049/TCP + ingress)
+      description = "NFS ingress from VPC private subnets"
+      cidr_blocks = module.vpc.private_subnets_cidr_blocks
+    }
+  }
+
+  tags = local.tags
+}
+
+# resource "aws_efs_file_system" "efs" {
+#   creation_token = "efs"
+#   encrypted      = true
+
+#   tags = local.tags
+# }
+
+# resource "aws_efs_mount_target" "efs_mt" {
+#   count = length(compact([for subnet_id, cidr_block in zipmap(module.vpc.private_subnets, module.vpc.private_subnets_cidr_blocks) : substr(cidr_block, 0, 4) == "100." ? subnet_id : null]))
+
+#   file_system_id  = aws_efs_file_system.efs.id
+#   subnet_id       = "${element(compact([for subnet_id, cidr_block in zipmap(module.vpc.private_subnets, module.vpc.private_subnets_cidr_blocks) : substr(cidr_block, 0, 4) == "100." ? subnet_id : null]), count.index)}"
+#   security_groups = [aws_security_group.efs.id]
+# }
+
+# resource "aws_security_group" "efs" {
+#   name        = "${local.name}-efs"
+#   description = "Allow inbound NFS traffic from private subnets of the VPC"
+#   vpc_id      = module.vpc.vpc_id
+
+#   ingress {
+#     description = "Allow NFS 2049/tcp"
+#     cidr_blocks = module.vpc.vpc_secondary_cidr_blocks
+#     from_port   = 2049
+#     to_port     = 2049
+#     protocol    = "tcp"
+#   }
+
+#   tags = local.tags
+# }
+
+resource "kubectl_manifest" "pv" {
+  yaml_body = <<YAML
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: efs-persist
+  namespace: ${local.name}
+spec:
+  capacity:
+    storage: 123Gi
+  accessModes:
+    - ReadWriteMany
+  nfs:
+    server: ${module.efs.dns_name}
+    path: "/"
+YAML
+
+  depends_on = [module.kubernetes_data_addons]
+}
+
+resource "kubectl_manifest" "pv" {
+  yaml_body = <<YAML
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: efs-persist
+  namespace: ${local.name}
+spec:
+  accessModes:
+    - ReadWriteMany
+  storageClassName: ""
+  resources:
+    requests:
+      storage: 10Gi
+YAML
+
+  depends_on = [module.kubernetes_data_addons]
+}
+
+#---------------------------------------------------------------
+# Cognito pool, domain and client creation.
+# This can be used
+# Auth integration later.
+#---------------------------------------------------------------
+
+resource "aws_cognito_user_pool" "pool" {
```

Comment:
@bryantbiggs Do we have any open source Cognito module for these resources?

Reply:
unfortunately, we do not have any Cognito modules at this time